### PR TITLE
README Update

### PR DESCRIPTION
**Pull request by Darshini-Parikh93**
All Component Sample related demo instructions are moved in the READMEs inside Demo folder. Top level README contains steps to perform demo using protocol samples and a link to Demo README for Component samples demo.

Added instructions to perform allowlist and denylist operations on RV component sample.

Added new APIs to the respective component REST contract.

Added instructions to enable remote access to DB.

Includes fix for github issues 74-79 and semantic fixes after proofread.

Signed-off-by: Darshini Parikh <darshini.k.parikh@intel.com>

### CODE REVIEW

File: demo/owner/README.md [hunk 1]
```diff
@@ -8,12 +8,122 @@ $ mvn clean install
...
+
+  Docker default: 300s
+
+- `owner_to0_rv_blob`
+
+  Information containing network address of prospective owner. Owner shares this information with RV during TO0. RV, then shares the same during TO1. Device, then uses this information to initiate TO2 protocol.
+
+  Docker default: http://localhost:8042?ipaddress=127.0.0.1
+
+- `owner_api_user'
```
Comment (prateekcr) [inline demo/owner/README.md:82]: Not rendered correctly. Last character is ' instead of `

File: README.md [hunk 2]
```diff
@@ -137,481 +137,4 @@ SELECT * FROM MT_DEVICES will show the current vouchers created by DI messages.
...
-  The URL at which the Device Initialization (DI) server may be found.
-
-  Default is `http://localhost:8039/`.
-
-- `fido.iot.pem.dev`
-
-  The location of the PEM file containing the device keys (private and public).
-  If not set, a hardcoded key is used - see the Java source for details.
-
-  There is no default. Provide value './device.pem' to use the existing default EC-256 key-pair.
```
Comment (prateekcr) [inline README.md:548]: Please add this line into device's README.

File: demo/README.md [hunk 3]
```diff
@@ -59,40 +57,60 @@ $ sudo docker rmi <image-id>
...
+
+The FIDO IoT Owner Sample currently supports the same `sdo_sys` module to send Owner service info to the Device and `devmod` module to store the received Device Service info. Few sample service info values, as present in \<fido-iot-src\>/demo/owner/serviceinfo/sample-values/ are populated into the database table 'OWNER_SERVICEINFO' as byte arrays. For keeping the association between the Device and the service info values to transfer, 'GUID_OWNERSVI' database table is used. When a Device is inserted into the database table 'TO2_DEVICES', it'll not have any association with the service info values, and so by default, no service info is transferred to the Device.
+
+# Enabling Service info transfer
+
+To enable service info transfer to a Device with a given GUID, following are the steps:
+
+1. (Optional) Insert required Service info values into the database table 'OWNER_SERVICEINFO' using the API `POST /api/v1/owner/svivalues/?id=<ServiceInfoId>&isCborEncoded=<boolean>`. More information about the same is provided in section [FIDO IoT Owner REST APIs](owner/README.md/#fido-iot-owner-rest-apis). If the required service info already exists in the table, go on to the next step.
```
Comment (prateekcr) [inline demo/README.md:114]: Missed out this in the earlier PR. Please update the API's supposed values <ServiceInfoId> & <boolean> as per Owner's REST APIs. Same for <guid> in the next point

File: demo/owner/README.md (hunk 1 above)
Comment (Darshini-Parikh93) [reply demo/owner/README.md]: Done

File: README.md (hunk 2 above)
Comment (Darshini-Parikh93) [reply README.md:548]: Done

File: demo/README.md (hunk 3 above)
Comment (Darshini-Parikh93) [reply demo/README.md]: Done

File: demo/README.md [hunk 4]
```diff
@@ -59,25 +57,30 @@ $ sudo docker rmi <image-id>
...
 
-Refer [Demo Manufacturer README](manufacturer/README.md) to start the service.
+```
+_JAVA_OPTIONS=-Dhttp.proxyHost=http_proxy_host -Dhttp.proxyPort=http_proxy_port -Dhttps.proxyHost=https_proxy_host -Dhttps.proxyPort=https_proxy_port
+```
+
+where
 
-# Start Reseller Service
+`http_proxy_host`: Represents the http proxy hostname. Typically, it is an IP address or domain name in the proxy URL.
```
Comment (trbehera) [inline demo/README.md:69]: the http proxy ==> the HTTP proxy

File: demo/README.md [hunk 5]
```diff
@@ -59,25 +57,30 @@ $ sudo docker rmi <image-id>
...
 
-Refer [Demo Reseller README](reseller/README.md) to start the service.
+`http_proxy_port`: Represents the http proxy port. Typically, it is the port number in the proxy URL.
```
Comment (trbehera) [inline demo/README.md:71]: the http proxy ==> the HTTP proxy

File: demo/README.md [hunk 6]
```diff
@@ -59,25 +57,30 @@ $ sudo docker rmi <image-id>
...
 
-# Start Rendezvous Service
+`https_proxy_host`: Represents the https proxy hostname. Typically, it is an IP address or domain name in the proxy URL.
 
-Refer [Demo Rendezvous README](rv/README.md) to start the service.
+`https_proxy_port`: Represents the https proxy port. Typically, it is the port number in the proxy URL.
 
-# Start Owner Service
+Specify combination the hostname and port information together for either http, https, or both. For example, if the http proxy is 'http://myproxy.com:900', then the following updates will be made to the properties:
```
Comment (trbehera) [inline demo/README.md:77]: Specify the combination of the hostname and the port information together ...

File: demo/README.md (hunk 4 above)
Comment (Darshini-Parikh93) [reply demo/README.md]: Done

File: demo/README.md (hunk 5 above)
Comment (Darshini-Parikh93) [reply demo/README.md]: Done

File: demo/README.md (hunk 6 above)
Comment (Darshini-Parikh93) [reply demo/README.md]: Done

**Review by trbehera — APPROVED** [review]

**Review by DukeDavis12 — DISMISSED** [review]

**Review by prateekcr — APPROVED** [review]

**Review by trbehera — APPROVED** [review]

**Review by prateekcr — APPROVED** [review]

**Pull request merged**